framework/src/main/java/org/checkerframework/framework/ajava/TypeAnnotationMover.java

```diff
@@ -0,0 +1,225 @@
+     * can't possibly be declaration annotations for that type of declaration.
+     *
+     * @param node JavaParser node for declaration
+     * @param declarationType the type of declaration {@code node} represents
+     * @return a list of annotations in declaration position that should be on the declaration's
+     *     type
+     */
+    private List<AnnotationExpr> getAnnotationsToMove(
+            NodeWithAnnotations<?> node, ElementType declarationType) {
+        List<AnnotationExpr> annosToMove = new ArrayList<>();
+        for (AnnotationExpr annotation : node.getAnnotations()) {
+            if (!isPossiblyDeclarationAnnotation(annotation, declarationType)) {
+                annosToMove.add(annotation);
+            }
+        }
+
+        return annosToMove;
+    }
+
+    /**
+     * Removes all annotations from {@code node} that appear in {@code annosToRemove}
+     *
+     * @param node a node with annotations
+     * @param annosToRemove list of annotations to remove
+     */
+    private void removeAnnotations(
+            NodeWithAnnotations<?> node, List<AnnotationExpr> annosToRemove) {
+        NodeList<AnnotationExpr> newAnnos = new NodeList<>(node.getAnnotations());
+        newAnnos.removeIf(anno -> annosToRemove.contains(anno));
+        node.setAnnotations(newAnnos);
+    }
+
+    /**
+     * Returns the TypeElement for an annotation, or null if it cannot be found.
+     *
+     * <p>If {@code annotation} was listed in the file's imports, returns its value in {@link
+     * #allAnnotations}. If it wasn't imported but its element could still be found, adds the new
+     * TypeElement to {@link #allAnnotations} and returns it.
+     *
+     * @param annotation a JavaParser annotation
+     * @return the TypeElement for {@code annotation}, or null if it cannot be found
+     */
+    private @Nullable TypeElement getAnnotationDeclaration(AnnotationExpr annotation) {
+        @SuppressWarnings("signature") // https://tinyurl.com/cfissue/3094
+        @FullyQualifiedName String annoNameFq = annotation.getNameAsString();
+        TypeElement annoTypeElt = allAnnotations.get(annoNameFq);
+        if (annoTypeElt == null) {
+            annoTypeElt = elements.getTypeElement(annoNameFq);
+            if (annoTypeElt == null) {
+                // Not a supported annotation.
+                return null;
+            }
+            AnnotationFileParser.putAllNew(
+                    allAnnotations,
+                    AnnotationFileParser.createNameToAnnotationMap(
+                            Collections.singletonList(annoTypeElt)));
+        }
+
+        return annoTypeElt;
+    }
+
+    /**
+     * Returns if {@code annotation} could be declaration annotation for {@code declarationType}.
+     * This would be the case if the annotation isn't recognized at all, or if it was recognized and
+     * has {@code declarationType} as one of its targets.
```

Comment:
If an annotation has no `@Target` meta-annotation, then it is applicable to declaration locations.  Please handle this case too.